### PR TITLE
Feature/connect api db

### LINKED ISSUE #17: Refactor frontend to ensure it works seamlessly with the backend

### PR DESCRIPTION
**Pull request by taranveersingh93**
This PR closes #17 

What I did:
Connect everything with a deployed backend/database

How I did it:
- I edited the URL's of the fetch requests. 
- Ensured that all tests are passing. Had to add one intercept of getting saved Tasks. 
- Added a helper function that humanizes the category data we receive from the db.

Did this break anything?
When I run `npx cypress open` it tells me it's not finding a tsConfig file and will skip some tsconfig-paths. I looked it up and from my understanding, it's not relevant to our project functionality. The tests will still run and the app will still function. I don't quite know where that could be causing troubles. Any directions would help too. 

- [x] No
- [ ]  Yes

Type of change

- [ ] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [x]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

Testing: 
Please pull this branch locally to see everything works as expected and check if all tests are passing.

### CODE REVIEW

File: src/components/SelectTaskView/SelectTaskView.js [hunk 1]
```diff
@@ -157,7 +160,7 @@ return (
     <div className="task-card">
       {tasksToShow === false && <h1>Loading...</h1>}
       {(tasksToShow && tasks.length !== 0 && !error.error) && <p className='task-text'>{currentTask.task}</p>}
-      {(!tasksToShow && !error.error) && <ErrorMessage />}
+      {(!tasksToShow && error.error) && <ErrorMessage />}
```
Comment (SethBurns) [inline src/components/SelectTaskView/SelectTaskView.js:163]: This error message is not showing up anymore now.

**Review by SethBurns — APPROVED** [review]
Approved pending error fix. Can merge this and then fix or fix and then merge. Either way. Nice work!

**Review by CaliHam — APPROVED** [review]
Checked out into branch, everything works great!

**Review by hvohr — APPROVED** [review]
Looks great !

File: src/components/SelectTaskView/SelectTaskView.js (hunk 1 above)
Comment (taranveersingh93) [reply src/components/SelectTaskView/SelectTaskView.js:163]: Hey @SethBurns Nice catch. I'll make another ticket for this and get right on it :)

**Pull request merged**